graph/graph.py

```diff
@@ -113,13 +113,18 @@ def update_links(self, links):
                 self.update_link_metadata(link)
 
     def update_link_metadata(self, link):
-        """Update link metadata."""
+        """Update link metadata."""        
         for key, value in link.metadata.copy().items():
             if key not in self._accepted_metadata:
                 continue
             endpoint_a = link.endpoint_a.id
             endpoint_b = link.endpoint_b.id
-            self.graph[endpoint_a][endpoint_b][key] = value
+            try:
+                self.graph[endpoint_a][endpoint_b][key] = value
+            except KeyError:
+                msg = ("Failed to get link from graph. Missing link"
+                       f" was from interfaces [{endpoint_a}, {endpoint_b}].")
+                raise LinkNotFound(msg)
```

Comment:
Side comment (if we get back to this PR in the future): we're always clearing the `graph` object before updating (although `topology` object is being updated correctly), so in a case where it never recovers (very unlikely unless we accidentally introduce a critical bug on core) we also end up with an incomplete and partial `graph`, but also rolling back the graph won't help much since that'd already be an older obsolete view of the topology. Ignoring also metadata could be something to consider and not crashing here, but at this point we'd also be accepting that a link endpoint also changed and we proceeded which can also lead to other major problems. So, yes, let's keep as it is, we're not making the application any worse (or increasing its risks) than it currently is